### PR TITLE
Simplify custom_extensions.py now that Aer and IQP aren't built

### PR DESCRIPTION
**Pull request by Eric-Arellano**
I was trying to figure out a docs issue in this repo and realized this could be simplified.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -43,7 +43,10 @@ commands =
 # Editable mode breaks macOS: https://github.com/sphinx-doc/sphinx/issues/10943
 usedevelop = false
 envdir = .tox/docs
-passenv = DOCS_FROM_MASTER QISKIT_DOCS_BUILD_TUTORIALS QISKIT_ENABLE_ANALYTICS
```
Comment (Eric-Arellano) [inline tox.ini:46]: Caused Tox 4 issues.

File: tox.ini [hunk 2]
```diff
@@ -57,10 +60,11 @@ commands =
 
 [testenv:docs-clean]
 skip_install = true
+deps =
```
Comment (Eric-Arellano) [inline tox.ini:63]: This makes the command faster.

File: tox.ini [hunk 3]
```diff
@@ -57,10 +60,11 @@ commands =
...
 allowlist_externals =
   rm
 commands =
-  rm -rf {toxinidir}/stubs/ {toxinidir}/docs/_build {toxinidir}/docs/apidoc {toxinidir}/docs/tutorials {toxinidir}/docs/plot_directive {toxinidir}/docs/jupyter_execute {toxinidir}/docs/.doctrees
+  rm -rf {toxinidir}/stubs/ {toxinidir}/docs/_build {toxinidir}/docs/stubs {toxinidir}/docs/apidoc {toxinidir}/docs/tutorials {toxinidir}/docs/migration_guides {toxinidir}/docs/plot_directive {toxinidir}/docs/jupyter_execute {toxinidir}/docs/.doctrees
```
Comment (Eric-Arellano) [inline tox.ini:67]: These folders are new since I added the command.

**Review by mtreinish — APPROVED** [review]
This seems straightforward enough

**Pull request merged**